src/Domain/CoordinateMaps/TimeDependent/Translation.cpp

```diff
@@ -96,42 +165,175 @@ std::array<tt::remove_cvref_wrap_t<T>, Dim> Translation<Dim>::frame_velocity(
+    }
+    radius.get() = sqrt(radius.get());
+
+    const DataVector function_of_time =
+        functions_of_time.at(f_of_t_name_)->func(time)[0];
+
+    auto result = make_with_value<
+        tnsr::Ij<tt::remove_cvref_wrap_t<DataType>, Dim, Frame::NoFrame>>(
+        dereference_wrapper(source_coords[0]), 0.0);
+    for (size_t i = 0; i < Dim; i++) {
+      for (size_t j = 0; j < Dim; j++) {
+        if constexpr (std::is_same_v<tt::remove_cvref_wrap_t<DataType>,
+                                     double>) {
+          if (radius.get() > 1.e-13) {
+            result.get(i, j) = (*f_of_r_).first_deriv(radius.get()) *
+                               gsl::at(function_of_time, i) *
+                               gsl::at(distance_to_center, j) / radius.get();
+          } else {
+            result.get(i, j) = (*f_of_r_).second_deriv(radius.get()) *
+                               gsl::at(function_of_time, i) *
+                               gsl::at(distance_to_center, j);
+          }
+        } else {
+          for (size_t k = 0; k < (radius.get()).size(); k++) {
+            if (radius.get()[k] > 1.e-13) {
+              result.get(i, j)[k] = (*f_of_r_).first_deriv(radius.get()[k]) *
+                                    gsl::at(function_of_time, i) *
+                                    gsl::at(distance_to_center, j)[k] /
+                                    radius.get()[k];
+            } else {
+              result.get(i, j)[k] = (*f_of_r_).second_deriv(radius.get()[k]) *
+                                    gsl::at(function_of_time, i) *
+                                    gsl::at(distance_to_center, j)[k];
+            }
+          }
+        }
+      }
+      result.get(i, i) += 1.0;
+    }
+    return result;
+  }
+}
+template <size_t Dim>
+template <typename DataType>
+tnsr::Ij<tt::remove_cvref_wrap_t<DataType>, Dim, Frame::NoFrame>
+Translation<Dim>::inv_jacobian(
+    const std::array<DataType, Dim>& source_coords, const double time,
+    const std::unordered_map<
+        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+        functions_of_time) const {
+  if (f_of_r_ == nullptr) {
+    return identity<Dim>(dereference_wrapper(source_coords[0]));
+  } else {
+    return determinant_and_inverse(
+               jacobian(source_coords, time, functions_of_time))
+        .second;
+  }
 }
 
 template <size_t Dim>
-template <typename T>
-tnsr::Ij<tt::remove_cvref_wrap_t<T>, Dim, Frame::NoFrame>
-Translation<Dim>::inv_jacobian(const std::array<T, Dim>& source_coords) const {
-  return identity<Dim>(dereference_wrapper(source_coords[0]));
+double Translation<Dim>::root_finder(const double& translated_radius,
+                                     const double& magnitude_function_of_time,
+                                     const double& root_factor) const {
+  const double center_offset = sqrt(magnitude_function_of_time);
+  double lower_bound = (translated_radius - center_offset) * (1.0 - 1.e-9);
```

Comment:
Can you add a comment about why you chose `+/-1.e-9`? Even if it's "this is what spec did".